### PR TITLE
8306940: test/jdk/java/net/httpclient/XxxxInURI.java should call HttpClient::close

### PR DESCRIPTION
**Pull request by dfuch**
The tests test/jdk/java/net/httpclient/*InURI.java create a lot of HttpClient instances.
They should call HttpClient::close when the client is no longer needed.

The fix includes some cleanup of the test code, as well as an additional check on the response version.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306940](https://bugs.openjdk.org/browse/JDK-8306940): test/jdk/java/net/httpclient/XxxxInURI.java should call HttpClient::close

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [9a143f3c](https://git.openjdk.org/jdk/pull/13673/files/9a143f3c1ff7a7da2c4329761f61cff8809df65b)
 * [Daniel Jeliński](https://openjdk.org/census#djelinski) (@djelinski - **Reviewer**) ⚠️ Review applies to [9a143f3c](https://git.openjdk.org/jdk/pull/13673/files/9a143f3c1ff7a7da2c4329761f61cff8809df65b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13673/head:pull/13673` \
`$ git checkout pull/13673`

Update a local copy of the PR: \
`$ git checkout pull/13673` \
`$ git pull https://git.openjdk.org/jdk.git pull/13673/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13673`

View PR using the GUI difftool: \
`$ git pr show -t 13673`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13673.diff">https://git.openjdk.org/jdk/pull/13673.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13673#issuecomment-1523693345)

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
Hello Daniel, these changes look fine to me.

File: test/jdk/java/net/httpclient/EscapedOctetsInURI.java [hunk 1]
```diff
@@ -121,120 +169,112 @@ void test(String uriString, boolean sameClient) throws Exception {
...
-    }
-
-    /** A handler that returns as its body the exact escaped request URI. */
-    static class HttpASCIIUriStringHandler implements Http2Handler {
+    static class HttpASCIIUriStringHandler implements HttpTestHandler {
         @Override
-        public void handle(Http2TestExchange t) throws IOException {
+        public void handle(HttpTestExchange t) throws IOException {
             String asciiUriString = t.getRequestURI().toASCIIString();
             out.println("Http2ASCIIUriString received, asciiUriString: " + asciiUriString);
```
Comment (djelinski) [inline test/jdk/java/net/httpclient/EscapedOctetsInURI.java:279]:
```suggestion
            out.println("HttpASCIIUriString received, asciiUriString: " + asciiUriString);
```

**Review by djelinski — APPROVED** [review]
LGTM.

**Pull request closed without merging**